sound/soc/sof/topology.c

```diff
@@ -1758,11 +1758,15 @@ static int sof_dai_unload(struct snd_soc_component *scomp,
 	struct snd_sof_pcm *spcm = dobj->private;
 
 	/* free PCM DMA pages */
-	if (spcm->pcm.playback)
+	if (spcm->pcm.playback) {
 		snd_dma_free_pages(&spcm->stream[SNDRV_PCM_STREAM_PLAYBACK].page_table);
+		kfree(spcm->stream[SNDRV_PCM_STREAM_PLAYBACK].time_info);
```

Comment:
You _need_ to set the time_info to NULL?

Reply:
sure, thanks

Reply:
spcm will be free after this so we don't need to